src/control/LatexController.cpp

```diff
@@ -184,44 +184,87 @@ void LatexController::showTexEditDialog()
 {
 	XOJ_CHECK_TYPE(LatexController);
 
-	LatexDialog* dlg = new LatexDialog(control->getGladeSearchPath());
+	dlg = new LatexDialog(control->getGladeSearchPath());
+	//For 'real time' LaTex rendering in the dialog
+	g_signal_connect(dlg->getTexBox(), "changed", G_CALLBACK(this->handleTexChanged), gpointer(this));
 	dlg->setTex(initalTex);
+	//The controller owns the tempRender because, on signal changed, he has to handle the old/new renders
+	insertTexImage(true);
+	dlg->setTempRender(temporaryRender->getImage());
 	dlg->show(GTK_WINDOW(control->getWindow()->getWindow()));
+	deletePreviousRender();
 	currentTex = dlg->getTex();
 	currentTex += " ";
 
 	delete dlg;
 }
 
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Text-changed handler: when the Entry in the dialog changes,
+ * this handler updates currentTex, removes the previous existing render and creates
+ * a new one. We need to do it through 'thisContr' because signal handlers
+ * cannot directly access non-static methods and non-static fields such as
+ * 'dlg' so we need to wrap all the dlg method inside small methods in 'thisContr'
+ * 
+ */
+void LatexController::handleTexChanged(GtkWidget *widget, gpointer data)
+{
+	LatexController *thisContr = ((LatexController *)data);
+
+	thisContr->setCurrentTex(gtk_entry_get_text(GTK_ENTRY(widget)));
+	thisContr->deletePreviousRender();
+	thisContr->runCommand();
+	thisContr->insertTexImage(true);
+	thisContr->setImageInDialog(thisContr->getTemporaryRender()->getImage());
+}
+
+TexImage* LatexController::getTemporaryRender(){
+	return this->temporaryRender;
+}
+
+void LatexController::setImageInDialog(cairo_surface_t *image)
+{
+	dlg->setTempRender(image);
+}
+
+void LatexController::deletePreviousRender()
+{
+	delete temporaryRender;
+}
+
+void LatexController::setCurrentTex(string currentTex)
+{
+	this->currentTex = currentTex;
+}
+
 void LatexController::deleteOldImage()
 {
 	XOJ_CHECK_TYPE(LatexController);
 
 	if (selectedTexImage)
 	{
-		EditSelection* selection = new EditSelection(control->getUndoRedoHandler(), selectedTexImage, view, page);
+		EditSelection *selection = new EditSelection(control->getUndoRedoHandler(), selectedTexImage, view, page);
```

Comment:
Usually we have the start at the type, not at the var name, as multi var declarations should not be used.